streams/src/main/java/org/apache/kafka/streams/errors/StreamsException.java

```diff
@@ -66,10 +80,15 @@ public StreamsException(final Throwable throwable, final TaskId taskId) {
      *          guarantee that the exception wasn't directly related to a specific task.
      */
     public Optional<TaskId> taskId() {
-        return Optional.ofNullable(taskId);
+        return taskIds.isEmpty() ? Optional.empty() : Optional.ofNullable(new ArrayList<>(taskIds).get(0));
     }
 
     public void setTaskId(final TaskId taskId) {
-        this.taskId = taskId;
+        taskIds.clear();
```

Comment:
I need to clear the tasks IDs otherwise you might not be able to get the task ID with `taskId()` after the first call to `setTaskId()`.